gammapy/estimators/utils.py

```diff
@@ -770,83 +773,350 @@ def combine_flux_maps(maps, method="gaussian_errors", reference_model=None):
-        sigma = Map.from_geom(geom, data=sigmas[0].data, unit=sigmas[0].unit)
-        gti = maps.gti
-        meta = maps.meta
-        if reference_model is None:
-            reference_model = maps.reference_model
+    gtis = [map_.gti for map_ in maps if map_.gti is not None]
+    if np.any(gtis):
+        gti = gtis[0].copy()
+        for k in range(1, len(gtis)):
+            gti.stack(gtis[k])
     else:
+        gti = None
+    # TODO : change this once we have stackable metadata objets
+    metas = [map_.meta for map_ in maps if map_.meta is not None]
+    meta = {}
+    if np.any(metas):
+        for data in metas:
+            meta.update(data)
+    if reference_model is None:
+        reference_model = maps[0].reference_model
+
+    if method == "gaussian_errors":
+
         means = [map_.dnde.copy() for map_ in maps]
         sigmas = [map_.dnde_err.copy() for map_ in maps]
+        # compensate for the ts deviation from gaussian approximation expectation in each map
+        ts_diff = np.nansum(
+            [
+                map_.ts.data - (map_.dnde.data / map_.dnde_err.data) ** 2
+                for map_ in maps
+            ],
+            axis=0,
+        )
+
         mean = means[0]
         sigma = sigmas[0]
 
-        gtis = [map_.gti for map_ in maps if map_.gti is not None]
-        if np.any(gtis):
-            gti = gtis[0].copy()
-            for k in range(1, len(gtis)):
-                gti.stack(gtis[k])
-        else:
-            gti = None
+        for k in range(1, len(means)):
+
+            mean_k = means[k].quantity.to_value(mean.unit)
+            sigma_k = sigmas[k].quantity.to_value(sigma.unit)
+
+            mask_valid = np.isfinite(mean) & np.isfinite(sigma) & (sigma.data != 0)
+            mask_valid_k = np.isfinite(mean_k) & np.isfinite(sigma_k) & (sigma_k != 0)
+            mask = mask_valid & mask_valid_k
+            mask_k = ~mask_valid & mask_valid_k
+
+            mean.data[mask] = (
+                (mean.data * sigma_k**2 + mean_k * sigma.data**2)
+                / (sigma.data**2 + sigma_k**2)
+            )[mask]
+            sigma.data[mask] = (
+                sigma.data * sigma_k / np.sqrt(sigma.data**2 + sigma_k**2)
+            )[mask]
+
+            mean.data[mask_k] = mean_k[mask_k]
+            sigma.data[mask_k] = sigma_k[mask_k]
+
+        ts = mean * mean / sigma / sigma + ts_diff
+
+        kwargs = dict(
+            sed_type="dnde", reference_model=reference_model, meta=meta, gti=gti
+        )
+        return FluxMaps.from_maps(dict(dnde=mean, dnde_err=sigma, ts=ts), **kwargs)
+
+    elif method in ["distrib", "profile"]:
+        if dnde_scan_axis is None:
+            dnde_scan_axis = _default_scan_map(maps[0]).geom.axes["dnde"]
+        for k, map_ in enumerate(maps):
+            map_stat_scan = (
+                interpolate_profile_map(map_, dnde_scan_axis=dnde_scan_axis)
+                if method == "profile"
```

Comment:
A line after the `if` is not missing here?? The syntax is surprising... 
Or I would write
```
if method != "profile:
  approximate_profile_map(map_, dnde_scan_axis=dnde_scan_axis)
```

Reply:
modified :
```
            if method == "profile":
                map_stat_scan = interpolate_profile_map(map_, dnde_scan_axis)
            else:
                map_stat_scan = approximate_profile_map(map_, dnde_scan_axis)

```